### DISCUSSION TITLE
Warning: CocoaPods is installed but broken. Skipping pod install.

### DISCUSSION THREAD

**Discussion by johanlantz** [codemagic-ci-cd/codemagic-docs#2091] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

My iOS builds have started to fail for no apparent reason today with this error:
```
Warning: CocoaPods is installed but broken. Skipping pod install.
  You appear to have CocoaPods installed but it is not working.
  This can happen if the version of Ruby that CocoaPods was installed with is different from the one being used to invoke it.
  This can usually be fixed by re-installing CocoaPods.
```
I tried logging in to the build machine with ssh and manually installing cocoapods with brew but it did not help.

Locally everything builds fine but the Codemagic builds have been failing for a couple of hours now. 

### Expected results

iOs build completes

### Actual results

Warning: CocoaPods is installed but broken. Skipping pod install.
  You appear to have CocoaPods installed but it is not working.
  This can happen if the version of Ruby that CocoaPods was installed with is different from the one being used to invoke it.
  This can usually be fixed by re-installing CocoaPods.
To re-install see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

CocoaPods not installed or not in valid state.

Build failed :|
Step 5 script `Make iOS build` exited with status code 1

### Build id (optional)

651ec635a0c14b8fce30ed99

**Comment by himesh-cm** [comment, marked as answer]
Hi @johanlantz,

It looks like the root cause is this - https://github.com/CocoaPods/CocoaPods/issues/12081

Can you please try the solution mentioned [here](https://github.com/CocoaPods/CocoaPods/issues/12081#issuecomment-1749571230).
